src/Analysis/Ast/Test/ClassesTests.cs

```diff
@@ -634,5 +634,14 @@ class Test():
             // Test run time: typically ~ 20 sec.
             sw.ElapsedMilliseconds.Should().BeLessThan(60000); 
         }
+
+        [TestMethod, Priority(0)]
+        public async Task IOErrorBase() {
+            const string code = @"
+class A(IOError): ...
```

Comment:
This example should be hitting the `_lines_with_members` function above I think.

The `str_from_typename` path is for an instance of the type, e.g.:

```python
class A: pass

GLOBAL_A = A(some, arguments, that, do, not, matter)
```

Becomes:

```
class A:
    """Documentation"""
    pass

GLOBAL_A = A()
```

Reply:
The problem was in `IOError = OSError()`  line in builtins which then prevented `RequestException` from having `IOError` as a base.

Reply:
Then there are two cases that both ought to be handled here. I guess when looking at the value we need to do an extra check to see if it's a direct alias or an instance of the type and set the scraped value appropriately.

Reply:
So the `IOError` case should be going through here:

https://github.com/microsoft/python-language-server/blob/a018586c98d2fceee23e8c6fea3502d56241df0b/src/Analysis/Ast/Impl/scrape_module.py#L714-L721

And other cases that end up in the same spot would go through here:

https://github.com/microsoft/python-language-server/blob/a018586c98d2fceee23e8c6fea3502d56241df0b/src/Analysis/Ast/Impl/scrape_module.py#L734-L736

Maybe add an extra flag to one or the other to control whether the parens appear (yes for the second case; no for the first)?

Reply:
So how's about in def __init__(

        if issubclass(value_type, type):
            self.instance = False
    def _str_from_typename(self, type_name):
        mod_name, sep, name = type_name.rpartition('.')
        s = self.name + ' = ' + safe_module_name(mod_name) + sep + name
        if self.instance:
            s = s + '()'
        return s